mkosi/__main__.py

```diff
@@ -8,29 +9,33 @@
 from .backend import MkosiException, die
 
 
-def main() -> None:
+@contextlib.contextmanager
+def propagate_failed_return():
     try:
-        args = parse_args()
-
-        for job_name, a in args.items():
-            # Change working directory if --directory is passed
-            if a.directory:
-                work_dir = a.directory
-                if os.path.isdir(work_dir):
-                    os.chdir(work_dir)
-                else:
-                    die(f"Error: {work_dir} is not a directory!")
-            if len(args) > 1:
-                with complete_step(f"Processing {job_name}"):
-                    run_verb(a)
-            else:
-                run_verb(a)
+        yield
     except MkosiException as e:
         cause = e.__cause__
         if cause and isinstance(cause, CalledProcessError):
             sys.exit(cause.returncode)
         sys.exit(1)
 
+@propagate_failed_return()
```

Comment:
Also another empty line for PEP8's sake
```suggestion

@propagate_failed_return()
```

Reply:
Actually no, PEP8 as far as I remember, specified a single empty line between functions. Two empty lines was between classes. Overall, I find this enforced formatting just a waste of time.

Reply:
It's [classes and top-level function definitions](https://peps.python.org/pep-0008/#blank-lines) and a single blank line between methods.